### PR TITLE
[docs] add submodule in engine.rst

### LINKED ISSUE #1459: Master engine docs missing something
## 🐛 Bug description

Latest master engine docs is missing some classes under `Engine` class, I believe.
Also `CallableEventWithFilter` link is not working.

Screen Shot 2020-11-12 at 20 13 22

### PR DESCRIPTION
**Pull request by ydcjeff**
Fixes #1459 

Description: Add submodules between classes in engine.rst

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

File: docs/source/engine.rst [hunk 1]
```diff
@@ -3,18 +3,24 @@ ignite.engine
...
+
 .. currentmodule:: ignite.engine.events
 
 .. autosummary::
     :nosignatures:
     :autolist:
 
+ignite.engine.deterministic
+
 .. currentmodule:: ignite.engine.deterministic
```
Comment (vfdev-5) [inline docs/source/engine.rst:25]: (could not comment on unmodified part)
```diff
- and helper methods:
+ and helper methods to define supervised trainer and evaluator:

File: docs/source/engine.rst [hunk 2]
```diff
@@ -3,18 +3,24 @@ ignite.engine
...
 
+ignite.engine.events
+
 .. currentmodule:: ignite.engine.events
 
 .. autosummary::
     :nosignatures:
     :autolist:
 
+ignite.engine.deterministic
```
Comment (vfdev-5) [inline docs/source/engine.rst:22]:
```suggestion
ignite.engine.deterministic (helper methods for deterministic training)
```

**Review by vfdev-5 — COMMENTED** [review]
PR looks good @ydcjeff ! I was just thinking to add few words explaining each submodule. I left an example in the comments.

File: docs/source/_templates/_static/css/ignite_theme.css [hunk 3]
```diff
@@ -131,8 +135,11 @@ article.pytorch-article table.longtable.docutils.align-default colgroup {
   display: none;
 }
 
-article.pytorch-article table.longtable.docutils.align-default tbody td:first-child {
-  width: 30%;
+article.pytorch-article
+  table.longtable.docutils.align-default
+  tbody
+  td:first-child {
+  width: 45%;
```
Comment (ydcjeff) [inline docs/source/_templates/_static/css/ignite_theme.css:142]: I only edit this `width`, the rest are formatted by pre-commit

**Comment by ydcjeff** [comment]
@vfdev-5 I found we can also add in contrib.engine and I added it. Also the table with width 30% doesn't look good in contrib.engine so I changed to 45%, the rest are formatted by pre-commit.

**Comment by vfdev-5** [comment]
@ydcjeff thanks for the updates ! When used 45% for tables, other tables start looking a bit different:
https://deploy-preview-1464--pytorch-ignite-preview.netlify.app/engine.html
vs 
https://pytorch.org/ignite/master/engine.html
Maybe,  a smaller value like 40% can be OK. @trsvchn what do you think ?

**Comment by ydcjeff** [comment]
@vfdev-5 yes that wide all applies to all `.. currentmodule`, maybe 40 looks good

**Comment by trsvchn** [comment]
@vfdev-5 Yes 45% is too much let's try 40%. Another option to specifically change that table using css selectors (if possible, I need to check this option)

**Review by vfdev-5 — APPROVED** [review]
Thanks !

**Pull request merged**